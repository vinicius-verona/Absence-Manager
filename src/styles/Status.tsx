import styled from "styled-components";

export const Label = styled.label`
  background-color: ${({ theme }) => theme.palettes.primary};
  border-radius: 15px;
  padding: 5px 10px;
  font-weight: 700;
  color: white;
`;
