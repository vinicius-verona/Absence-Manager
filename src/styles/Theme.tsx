// eslint-disable-next-line import/no-anonymous-default-export
export default {
    palettes: {
        primary: "rgba(255, 148, 25, 1)",
        lightPrimary: "rgba(255, 148, 25, 0.5)",
        light: "#DFDFDF",
        darkerLight: "#B5B5B5",
        bg_dark: "rgba(0, 0, 0, 0.5)",
    },
};
