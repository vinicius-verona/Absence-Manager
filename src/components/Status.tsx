import { Label } from "../styles/Status";

type StatusContent = {
  content: string;
};

/**
 * The `Status` component returns a status with its content styled in order to stand-out.
 * @param props.content - the status value
 */
export default function Status(props: StatusContent) {
  return (
    <div>
      <label>
        <strong>Status: </strong>
      </label>
      <Label>{props.content}</Label>
    </div>
  );
}
