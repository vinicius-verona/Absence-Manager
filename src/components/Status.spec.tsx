import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import renderer from "react-test-renderer";
import Theme from "../styles/Theme";
import Status from "./Status";

describe("Render absence' status", () => {
  it("Should match status snapshot", () => {
    const status = renderer
      .create(
        <ThemeProvider theme={Theme}>
          <Status content={"Confirmed"} />
        </ThemeProvider>
      )
      .toJSON();

    expect(status).toMatchSnapshot();
  });

  it("Should render label", () => {
    const { getByText } = render(
      <ThemeProvider theme={Theme}>
        <Status content={"Confirmed"} />
      </ThemeProvider>
    );

    expect(getByText("Status:")).toBeInTheDocument();
  });

  it("Should render status content", () => {
    const { getByText } = render(
      <ThemeProvider theme={Theme}>
        <Status content={"Confirmed"} />
      </ThemeProvider>
    );

    expect(getByText("Confirmed")).toBeInTheDocument();
  });
});
