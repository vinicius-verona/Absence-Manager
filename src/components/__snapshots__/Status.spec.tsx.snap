// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render absence' status Should match status snapshot 1`] = `
<div>
  <label>
    <strong>
      Status: 
    </strong>
  </label>
  <label
    className="sc-bdvvtL hKoNsd"
  >
    Confirmed
  </label>
</div>
`;
